### PR TITLE
Test authz helper

### PR DESCRIPTION
**Pull request by fntlnz**
giphy

Signed-off-by: Lorenzo Fontana <fontanalorenzo@me.com>

### CODE REVIEW

File: authorization/api_test.go [hunk 1]
```diff
@@ -1,7 +1,138 @@
...
+
+type TestPlugin struct {
+	Plugin
+}
+
+func (p *TestPlugin) AuthZReq(r Request) Response {
+	return Response{
+		Allow: false,
+		Msg:   "You are not authorized",
+		Err:   "",
```
Comment (runcom) [inline authorization/api_test.go]: Would be fine have this scenario tested either.

**Comment by runcom** [comment]
Ping @vdemeester also, would be cool to have your "for range" loop with each scenario (as you did in docker/docker time ago) :)  any idea?

File: authorization/api_test.go (hunk 1 above)
Comment (fntlnz) [inline authorization/api_test.go]: Which one?

Comment (runcom) [inline authorization/api_test.go]: Err not empty (somehow)

Comment (fntlnz) [inline authorization/api_test.go]: I can integrate with a few more cases. 

Allow: true, Msg: "", Err: ""
Allow: false, Msg: "", Err:""
Allow: false, Msg: "", Err: "i just had an error"

**Comment by fntlnz** [comment]
@runcom Let me know if there are more cases/things that need to be tested here, thanks!

**Comment by vdemeester** [comment]
@runcom @fntlnz you could add some attribute to the `TestPlugin` struct to be able to customize the response for tests (and thus have a slice & co) :angel:

**Comment by vdemeester** [comment]
ping @runcom it's also ok to merge this as is and refactor test after 😇

**Comment by runcom** [comment]
@vdemeester fine, we're just testing a subset of all the scenario though :)

**Comment by fntlnz** [comment]
@vdemeester @runcom Just wait a few hours, I'll integrate with the other things :) If I understood well you want to make `TestPlugin` configurable in order to test different use cases right?

**Comment by vdemeester** [comment]
@runcom yeah more test == better, even if it's not complete :angel: 
@fntlnz yes :+1:

**Comment by dave-tucker** [comment]
@fntlnz hi! any update on this PR?
If not, can we take take @vdemeester's advice and merge now, refactor later?
/cc @runcom

**Comment by vdemeester** [comment]
I do agree with @dave-tucker, let's merge this now and refactor later

**Comment by runcom** [comment]
LGTM then, @vdemeester feel free to merge!

**Comment by vdemeester** [comment]
@runcom I don't have the writing rights on the repository 👼

**Pull request merged**